File: modules/cubex-core/src/main/java/org/cubexmc/core/Messager.java
package org.cubexmc.core;

import java.util.Collections;
import org.bukkit.command.CommandSender;

public final class Messager {

    public void send(CommandSender target, String message) {
        if (target == null || message == null) {
            return;
        }
        String[] lines = message.split("\\R", -1);
        for (String line : lines) {
            target.sendMessage(line);
        }
    }

    public void sendLines(CommandSender target, Iterable<String> messages) {
        if (target == null) {
            return;
        }
        for (String message : messages == null ? Collections.<String>emptyList() : messages) {
            send(target, message);
        }
    }
}

File: modules/cubex-core/src/main/kotlin/org/cubexmc/core/Messager.kt
package org.cubexmc.core

import java.util.regex.Pattern
import org.bukkit.command.CommandSender

class Messager {
    fun send(target: CommandSender?, message: String?) {
        if (target == null || message == null) return

        for (line in LINE_BREAK.split(message, -1)) {
            target.sendMessage(line)
        }
    }

    fun sendLines(target: CommandSender?, messages: Iterable<String>?) {
        if (target == null) return

        for (message in messages ?: emptyList()) {
            send(target, message)
        }
    }

    private companion object {
        val LINE_BREAK: Pattern = Pattern.compile("\\R")
    }
}
